cmd/ipfs/main.go

```diff
@@ -315,6 +315,13 @@ func callCommand(ctx context.Context, req cmds.Request, root *cmds.Command) (cmd
 		return nil, err
 	}
 
+	if cmd.PreRun != nil {
+		err = cmd.PreRun(req)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
That's kind of hacky for `PreRun` since we would have to generate a fake `Response` to pass it. And if it set anything on that fake response, we would have to somehow combine that state with the real one.

But we can at least make `PostRun` the same as the signature for `Run`.